src/rendergraph/opengl/backend/shadercache.h

```diff
@@ -0,0 +1,48 @@
+#pragma once
+
+#include <map>
+
+#include "rendergraph/material.h"
+#include "rendergraph/materialshader.h"
+#include "rendergraph/materialtype.h"
+
+namespace rendergraph::backend {
+class ShaderCache;
+}
+
+class rendergraph::backend::ShaderCache {
+  private:
+    static std::map<rendergraph::MaterialType*,
+            std::shared_ptr<rendergraph::MaterialShader>>&
+    map() {
+        static std::map<rendergraph::MaterialType*,
+                std::shared_ptr<rendergraph::MaterialShader>>
+                s_map;
+        return s_map;
+    }
+
+  public:
+    static std::shared_ptr<rendergraph::MaterialShader> getShaderForMaterial(
+            rendergraph::Material* pMaterial) {
+        auto iter = map().find(pMaterial->type());
+        if (iter != map().end()) {
+            return iter->second;
+        }
+        auto pResult = std::shared_ptr<rendergraph::MaterialShader>(
+                pMaterial->createShader().release());
```

Comment:
you can create a shared_ptr from the unique_ptr directly.
```suggestion
        auto pResult = std::shared_ptr<rendergraph::MaterialShader>(
                pMaterial->createShader());
```